conda_lock/invoke_conda.py

```diff
@@ -116,14 +117,21 @@ def _invoke_conda(
     ) as p:
         stdout = []
         if p.stdout:
-            for line in _process_stdout(p.stdout):
-                logging.info(line)
-                stdout.append(line)
+
+            def read_stdout() -> None:
+                assert p.stdout is not None
+                for line in _process_stdout(p.stdout):
+                    logging.info(line)
+                    stdout.append(line)
+
+            stdout_thread = threading.Thread(target=read_stdout)
+            stdout_thread.start()
         stderr = []
         if p.stderr:
             for line in p.stderr:
                 stderr.append(line)
                 logging.error(line.rstrip())
+        stdout_thread.join()
```

Comment:
```suggestion
        if p.stdout:
            stdout_thread.join()
```

Alternatively, we can `stdout_thread = None` before the first `if p.stdout:`, and check `if stdout_thread:` here.